backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -110,6 +110,24 @@ export class SegmentService {
         });
       }
 
+      segmentsData.forEach((segment) => {
+        segmentsUsedLockedList.forEach((usedSegment) => {
+          if (segment.id === usedSegment) {
```

Comment:
We can replace this third-level loop with something like:          `segmentsUsedLockedList.push(...segment.subSegments.map(subSegment => subSegment.id))`
and also below for the unlocked list.

Reply:
the second level loop can be replaced with an if condition
`if(segmentUsedList.includes(segment.id))`

Reply:
resolved